cmd/buildah/rmi.go

```diff
@@ -43,55 +57,38 @@ func rmiCmd(c *cli.Context) error {
 		_, err = store.DeleteImage(id, true)
 		if err != nil {
 			var ref types.ImageReference
-			// If it's looks like a proper image reference, parse
-			// it and check if it corresponds to an image that
-			// actually exists.
-			if ref2, err2 := alltransports.ParseImageName(id); err2 == nil {
-				if img, err3 := ref2.NewImage(nil); err3 == nil {
-					img.Close()
-					ref = ref2
-				} else {
-					logrus.Debugf("error confirming presence of image %q: %v", transports.ImageName(ref2), err3)
-				}
-			} else {
-				logrus.Debugf("error parsing %q as an image reference: %v", id, err2)
+			ref, err2 := properImageRef(id)
+			if err2 != nil {
+				logrus.Debug(err2)
 			}
 			if ref == nil {
-				// If it's looks like an image reference that's
-				// relative to our storage, parse it and check
-				// if it corresponds to an image that actually
-				// exists.
-				if ref2, err2 := storage.Transport.ParseStoreReference(store, id); err2 == nil {
-					if img, err3 := ref2.NewImage(nil); err3 == nil {
-						img.Close()
-						ref = ref2
-					} else {
-						logrus.Debugf("error confirming presence of image %q: %v", transports.ImageName(ref2), err3)
-					}
-				} else {
-					logrus.Debugf("error parsing %q as a store reference: %v", id, err2)
+				if ref, err2 = storageImageRef(store, id); err2 != nil {
+					logrus.Debug(err2)
 				}
 			}
 			if ref == nil {
-				// If it might be an ID that's relative to our
-				// storage, parse it and check if it
-				// corresponds to an image that actually
-				// exists.  This _should_ be redundant, since
-				// we already tried deleting the image using
-				// the ID directly above, but it can't hurt,
-				// either.
-				if ref2, err2 := storage.Transport.ParseStoreReference(store, "@"+id); err2 == nil {
-					if img, err3 := ref2.NewImage(nil); err3 == nil {
-						img.Close()
-						ref = ref2
-					} else {
-						logrus.Debugf("error confirming presence of image %q: %v", transports.ImageName(ref2), err3)
-					}
-				} else {
-					logrus.Debugf("error parsing %q as an image reference: %v", "@"+id, err2)
+				if ref, err2 = storageImageID(store, id); err2 != nil {
+					logrus.Debug(err2)
 				}
 			}
 			if ref != nil {
+				if force {
+					// Remove all running containers matching ref
+					image, err2 := is.Transport.GetImage(ref)
+					if err2 != nil {
+						logrus.Debugf("Error parsing image ID: %v\n", err2)
+					}
+					containers, err2 := store.Containers()
+					if err2 != nil {
+						logrus.Debugf("Error getting associated containers: %v\n", err)
+					}
+					for i := 0; i < len(containers); i++ {
+						if containers[i].ImageID == image.ID {
+							fmt.Printf("Found the container")
```

Comment:
Not real helpful to have 1 to n "Found the container" 's.  If you think it's important to note, perhaps change this to something like fmt.Printf("Deleting container %s", containers[i].ID)